### ISSUE TITLE
Aggregation for multiple months is grabbing data from last month

### ISSUE THREAD

**Issue by AzureEnigma** [cloudfoundry-incubator/cf-abacus#225]
When grabbing the aggregation report for February with a SLACK window of 27D, the aggregator not only grabs data from February, but also aggregates data from January as well.

We are seeing customers being aggregated from last month. Their usages are coming from the last month aggregation as data that has been consumed already.

``` javascript
"consumers": [
        "cloud-foundry-application:***/t/0001453841048087-0-0-1-0",
        "cloud-foundry-application:***/t/0001453841048084-0-0-1-0",
        "cloud-foundry-application:***/t/0001453841046997-0-0-1-0",
        "cloud-foundry-application:***/t/0001453841047310-0-0-1-0",
        "cloud-foundry-application:***/t/0001453841048080-0-0-1-0",
        "cloud-foundry-application:***/t/0001453841047374-0-0-1-0",
        "cloud-foundry-application:***/t/0001453841047156-0-0-1-0",
        "cloud-foundry-application:***/t/0001453841047093-0-0-1-0",
        "cloud-foundry-application:/t/0001453840828488-0-0-1-0",
        "cloud-foundry-application:***/t/0001453841047527-0-0-1-0",
        "cloud_foundry_application:***/t/0001453859234603-1-0-1-0"
      ]
```
This was the aggregation report was for Mon, 01 Feb 2016 18:45:55 GMT. This was the first report for the month of February, so the consumption should be nothing, however, we see that stuff has already been consumed.

(This was the current month window (for February))

```javascript
{
                    "quantity": {
                      "consumed": 5428455250,
                      "consuming": 89.25,
                      "since": 1452300257000
                    },
                    "cost": {
                      "burned": 0,
                      "burning": 0,
                      "free": 0,
                      "since": 1452300257000
                    }
                  }
```

``` javascript
"start": 1452038400000,
  "end": 1453132538338,
  "account_id": "***",
  "id": "k/***/t/0001454352355707-0-0-3-0",
  "processed": 1454352355707
```

**Comment by BetaFood** [comment]
There's a couple of things here.

The aggregator should be starting off fresh in a new month with the default configuration. Since there is no previous "aggregated" document, it should pretty much start empty.

The datamodel refactor that happened a little bit ago makes this a bit more complicated. If we do get a new aggregated usage for a new month, the connections to the previous month's consumers would be lost(assuming a slack is defined).

In the case of the usage from the previous month carrying over into the current month, the consumer connections are still there, but they're outdated. For instance, if one consumer has no usage in the current month. When the report retrieves it, it will show the previous month's consumer report as of it was the current month.
```JSON
{
  "windows": [null, null, null, null,
    [{"quantity": 20}, null]
  ]
}
```
for instance would be return as is in both the current and previous month if no further usage for that consumer comes in during the current month.

So in addition to the problem with the aggregator reducing off of the previous month, I believe two other problems are:
* In a clean aggregated document for a new month, in the case of a slack, the aggregated usage may need to point to the previous month's consumers
* In a transferred aggregated document for a new month(current case), in the case of a slack, the aggregated usage needs to either filter out the slack usage or push the slack usage so that it makes sense with the report's query time.

**Comment by jsdelfino** [comment]
Is that fixed now? I thought so but the issue is still open.

**Comment by BetaFood** [comment]
This should be solved by 41de534ca25a2bb84e538cb8c3f7e6810294f357 and 1514b73a77a89151d4fe048b484bd037c6cbe24c. There's a little harmless effect of having outdated consumer in the report with no usage, but it shouldn't give wrong usage like before.

If there's definite issues in levels above the consumer, feel free to open a new issue or reopen this one.
